### PR TITLE
feat:(composables): Add `createUseDevice` composable factory.

### PR DESCRIPTION
**Pull request by tajespasarela**
EX-8122

### CODE REVIEW

File: packages/x-components/src/composables/create-use-device.composable.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+/**
+ * Util function to capitalize a string.
+ *
+ * @param str - The string to capitalize.
+ * @returns The string capitalized.
+ *
+ * @internal
+ */
+function capitalize(str: string): string {
```
Comment (tiborux) [inline packages/x-components/src/composables/create-use-device.composable.ts:204]: We already have a util for this inside x-components.

File: packages/x-components/src/composables/create-use-device.composable.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+/**
+ * Factory function that creates a composable for device detection using the devices parameter
+ * to configure breakpoints.
+ *
+ * @param devices - An object containing the breakpoints, where the key is the name of the device
+ * and the value is the screen width.
+ * @returns A composable which provides multiple reactive flags and values for detecting the
+ * current device. The flags names depends on the names passed in the `devices` parameter.
+ * @remarks The `orientation` only works for orientation- sensor devices (mobile, tablet, etc). If
```
Comment (CachedaCodes) [inline packages/x-components/src/composables/create-use-device.composable.ts:36]:
```suggestion
 * @remarks The `orientation` only works for orientation-sensor devices (mobile, tablet, etc). If
```

File: packages/x-components/tests/unit/useDevice.spec.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    getDeviceValue(key: string) {
+      return cy.getByDataTest(key);
+    }
+  };
+}
+
+const defaultDevices = { mobile: 0, tablet: 744, desktop: 1200 };
+describe('testing createUseDevice composable factory', () => {
+  it('detects the right device', () => {
+    const { getDeviceValue } = mountTestComponent(defaultDevices);
```
Comment (tiborux) [inline packages/x-components/tests/unit/useDevice.spec.ts:41]: Should we test the orientation? I know that we are using the composable from vue use, but we should check our own values

File: packages/x-components/src/composables/create-use-device.composable.ts [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+  return reduce(
+    devices,
+    (accumulator, device, deviceWidth) => {
+      const isDevice = computed(
+        () =>
+          deviceWidth <= windowSize.value &&
+          !sortedByWidthDevices.some(
+            ([, otherDeviceWidth]) =>
+              otherDeviceWidth <= windowSize.value && otherDeviceWidth > deviceWidth
+          )
```
Comment (CachedaCodes) [inline packages/x-components/src/composables/create-use-device.composable.ts:109]: Can we extract this to its own function? I feel like `getDeviceFlags` does a lot already

**Review by CachedaCodes — CHANGES_REQUESTED** [review]
My head hurts, gj!

File: packages/x-components/tests/unit/useDevice.spec.ts (hunk 3 above)
Comment (tajespasarela) [reply packages/x-components/tests/unit/useDevice.spec.ts:41]: I would like, and I tried, but unfortunately, this feature only works on actual mobile/tablet/with-gyroscope devices. with cypress, you only can modify the size of the window but it always returns the "landscape" value, even in a "portrait" size.

File: packages/x-components/src/composables/create-use-device.composable.ts (hunk 1 above)
Comment (tajespasarela) [reply packages/x-components/src/composables/create-use-device.composable.ts:204]: I was looking for it in the `x-utils` package. We have to move it there in the near future.

File: packages/x-components/src/composables/create-use-device.composable.ts [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+ *
+ * @returns A reactive value indicating the current device
+ * orientation.
+ *
+ * @internal
+ */
+function getOrientation(): UseDeviceReturn['orientation'] {
+  const { orientation } = useScreenOrientation();
+  return computed(() => {
+    if (['landscape-primary', 'landscape-secondary'].includes(orientation.value ?? '')) {
```
[the remark this answers is not in the document]
Comment (tajespasarela) [reply packages/x-components/src/composables/create-use-device.composable.ts:135]: done

File: packages/x-components/src/composables/create-use-device.composable.ts (hunk 2 above)
Comment (tajespasarela) [reply packages/x-components/src/composables/create-use-device.composable.ts:36]: changed

File: packages/x-components/src/composables/create-use-device.composable.ts [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+ *
+ * @internal
+ */
+function getDeviceName(
+  devices: Record<string, number>,
+  devicesFlags: UseDeviceFlags<string>
+): Ref<string> {
+  return computed(
+    () =>
+      Object.keys(devices).find(device => devicesFlags[`is${capitalize(device)}` as any]?.value) ??
```
Comment (tiborux) [inline packages/x-components/src/composables/create-use-device.composable.ts:180]: I was trying to remove this any ⚰️

Comment (tajespasarela) [reply packages/x-components/src/composables/create-use-device.composable.ts:180]: done

File: packages/x-components/src/composables/create-use-device.composable.ts (hunk 4 above)
Comment (tajespasarela) [reply packages/x-components/src/composables/create-use-device.composable.ts:109]: done

**Review by tiborux — APPROVED** [review]
gj

**Review by CachedaCodes — APPROVED** [review]

**Pull request merged**